website/docs/pact_nirvana/step_5.md

```diff
@@ -1,32 +1,78 @@
+## Tasks
+
+### A. Publish the pact from your consumer PR pipeline
+
+If you have done it correctly, then the consumer contract tests should run as part of your regular test run. But you still need to modify your PR validation job to publish the pact, correctly identifying the consumer version and apply a tag with the PR branch name
+
+1. Configure your consumer build to run the Pact tests and publish its pact to the Broker as part of its main build \(consult the documentation for your chosen language\). Consumer Pact tests typically run after the unit tests, and before deploying to a test environment.
+2. Configure a tag name to be used for every consumer build that publishes a pact (again, see your Pact language docs). The recommended default is to dynamically determine and use the name of your git/svn branch. If this doesn’t work for you, then you could hardcode it to something like "master" or "stable". It will make the configuration simpler if each consumer and provider uses the same tag name for the stable "main line" of development, so it's best to decide on this together.
+
+### B. Modify the consumer commit pipeline to tag with master branch
+
+After you have set up the consumer PR pipeline to publish pacts, you need to also modify your consumer's main 
+branch commit pipeline to indicate that a particular consumer version is now in the main branch.
+
+### C. Add a step at the beginning of the pipeline to apply the tag with the name of your main branch (e.g. master, trunk or main)
+
+The provider verification needs this information so it can get the latest pact that has been committed to the main 
+branch. If it gets the latest pact, it may get a pact that is committed on a feature branch and not ready for verification.
+
+### D. Configure pact to be verified when provider changes
+
+One the consumer is publishing its pacts and is tagging them with the correct branch names, you can now add Pact verification and to your provider PR pipeline.
+
+Pact verification should run as part of your regular unit test run. But you should change the tag it verifies against to be the name of the consumer's main branch instead of `latest`.
+
+1. Configure your provider build to fetch the pact(s) from the broker and publish the verification results as part of its main build \(consult the documentation for your chosen language\). This would typically happen after the unit tests, and before deploying to a test environment. You can find the recommended configuration [here](/provider/recommended_configuration#verification-triggered-by-provider-change).
+2. In the provider verification configuration, change the pact that is being verified from the latest pact to the latest pact for the stable tag (see the relevant documentation for your library). This will help keep your provider builds green.
+
+
+### Feature Flag notes
```

Comment:
I think maybe don't include this here. This is supposed to give you a high-level roadmap, and we don't want to wander too much into the details. Maybe instead add this to the existing doc on feature toggles?

Reply:
Agreed, So this was taken from https://docs.pact.io/pact_nirvana/step_5 

I tried to pull out but not lose the feature flag bits in the merge, but agree they are best off in the honours course :)

This line and other mentions to feature flag are worth pulling out. I know you've started some discussion about feature flags in general which is valuable and worth incorporating or revisiting entirely to gain more clarity.

https://github.com/pact-foundation/docs.pact.io/pull/128/files#diff-5e838a062066c7a67db8e2c20c6e37245d118b9a28187da841989938a341a830L22

Reply:
Hey. You keep providing links to specific diffs. This is not something I've seen before in github reviews. I could use a little education how to use that. Thanks!

Reply:
Certainly for linking lines see below

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet

For viewing line based history

https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#viewing-the-line-by-line-revision-history-for-a-file

I use them often in code reviews when I am in unfamiliar territory and try to understand the rational for a change, at a point in time :)

Right signing off now, possibly for the weekend (it's pretty late here 😅) but will catch up with the rest of the comments soon

Reply:
Moved out now, will capture in your feature flags piece if required